### DISCUSSION TITLE
Converting Custom Deserializations from 3.X to 5.X

### DISCUSSION THREAD

**Discussion by mfonnesbeck** [ravendb/ravendb#16727] [Q&A]
We are upgrading our database version and I'm having a tough time converting over our custom Json deserializers. We previously set the CustomizeJsonSerializer, as part of the newly setup document store, to a method that added converters to the serializer. One converter would read the Json being deserialized and would properly create a new Security Claim value for roles for users. But looking over articles that talk about setting up the custom serializer and deserializers is not calling the read method within the Claim converter and reading the values from Raven.

This is what I have so far:
DocumentStore store = new DocumentStore { 
                Conventions = { 
                    Serialization = new CustomSerializationConventions(),
                }
};
container.RegisterSingleton<IDocumentStore>(store);
store.Initialize();

public class CustomSerializationConventions : NewtonsoftJsonSerializationConventions
{
    public CustomSerializationConventions()
    {
        CustomizeJsonSerializer = CustomizeSerializer;
        CustomizeJsonDeserializer = CustomizeSerializer;
    }

    private static void CustomizeSerializer(JsonSerializer serializer)
    {
        serializer.Converters.Add(new ClaimConverter());
    }
}

public class ClaimConverter : JsonConverter
{
    public override bool CanWrite
    {
        get
        {  return false; }
    }

    public override bool CanConvert(Type objectType)
    {
        return objectType == typeof(Claim);
    }

    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
    {
        JObject jObject = JObject.Load(reader);
        return new Claim(jObject["Type"].Value<String>(), jObject["Value"].Value<string>());
    }

    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
    {
        return;
    }
}

The Claim class is a standard Microsoft .Net class from System.Security.Claims.

**Comment by ayende** [comment]
You should be able to just use:

```
store.Conventions.Serialization = new NewtonsoftJsonSerializationConventions
                {
                    CustomizeJsonSerializer = serializer  => serializer.Converters.Add(new ClaimConverter()),
                    CustomizeJsonDeserializer = serializer  => serializer.Converters.Add(new ClaimConverter()),
                }
```
In the initialization

If this doesn't work, can you create a small test case?
